packages/fast_time/src/pal/facade/time_source.rs

```diff
@@ -49,7 +52,10 @@ impl TimeSource for TimeSourceFacade {
             #[cfg(any(miri, not(any(target_os = "linux", windows))))]
             Self::Passthrough(ts) => ts.now(),
             #[cfg(test)]
-            Self::Mock(ts) => ts.now(),
+            Self::Mock(ts) => ts
+                .lock()
+                .expect("mock time source lock should not be poisoned")
```

Comment:
I am not convinced that this conforms to the repository guidelines about expect() messages. Verify that it does.